src/meta/test/BalancerTest.cpp

```diff
@@ -0,0 +1,548 @@
+#include <gtest/gtest.h>
+#include <folly/executors/CPUThreadPoolExecutor.h>
+#include <folly/synchronization/Baton.h>
+#include "meta/processors/admin/Balancer.h"
+#include "meta/test/TestUtils.h"
+#include "fs/TempDir.h"
+#include "meta/processors/partsMan/CreateSpaceProcessor.h"
+
+namespace nebula {
+namespace meta {
+
+class TestFaultInjector : public FaultInjector {
+public:
+    explicit TestFaultInjector(std::vector<Status> sts)
+        : statusArray_(std::move(sts)) {
+        executor_.reset(new folly::CPUThreadPoolExecutor(1));
+    }
+
+    ~TestFaultInjector() {
+    }
+
+    folly::Future<Status> response(int index) {
+        folly::Promise<Status> pro;
+        auto f = pro.getFuture();
+        executor_->add([this, p = std::move(pro), index]() mutable {
+            p.setValue(this->statusArray_[index]);
+        });
+        return f;
+    }
+
+    folly::Future<Status> transLeader() override {
+        return response(0);
+    }
+
+    folly::Future<Status> addPart() override {
+        return response(1);
+    }
+
+    folly::Future<Status> addLearner() override {
+        return response(2);
+    }
+
+    folly::Future<Status> waitingForCatchUpData() override {
+        return response(3);
+    }
+
+    folly::Future<Status> memberChange() override {
+        return response(4);
+    }
+
+    folly::Future<Status> updateMeta() override {
+        return response(5);
+    }
+
+    folly::Future<Status> removePart() override {
+        return response(6);
+    }
+
+    void reset(std::vector<Status> sts) {
+        statusArray_ = std::move(sts);
+    }
+
+private:
+    std::vector<Status> statusArray_;
+    std::unique_ptr<folly::Executor> executor_;
+};
+
+TEST(BalanceTaskTest, SimpleTest) {
+    {
+        std::vector<Status> sts(7, Status::OK());
+        std::unique_ptr<FaultInjector> injector(new TestFaultInjector(std::move(sts)));
+        auto client = std::make_unique<AdminClient>(std::move(injector));
+        BalanceTask task(0, 0, 0, HostAddr(0, 0), HostAddr(1, 1), nullptr, nullptr);
+        folly::Baton<true, std::atomic> b;
+        task.onFinished_ = [&]() {
+            LOG(INFO) << "Task finished!";
+            EXPECT_EQ(BalanceTask::Result::SUCCEEDED, task.ret_);
+            EXPECT_EQ(BalanceTask::Status::END, task.status_);
+            b.post();
+        };
+        task.onError_ = []() {
+            LOG(FATAL) << "We should not reach here!";
+        };
+        task.client_ = client.get();
+        task.invoke();
+        b.wait();
+    }
+    {
+        std::vector<Status> sts{Status::Error("transLeader failed!"),
+                                Status::OK(),
+                                Status::OK(),
+                                Status::OK(),
+                                Status::OK(),
+                                Status::OK(),
+                                Status::OK()};
+        std::unique_ptr<FaultInjector> injector(new TestFaultInjector(std::move(sts)));
```

Comment:
We can't use make_unique in this case.  Because the unique_ptr is for FaultInjector, but it takes pointer from TestFaultInjector.  There is an implicit convert inside it.